aten/src/TH/generic/THTensorEvenMoreMath.cpp

```diff
@@ -221,8 +221,12 @@ void THTensor_(indexSelect)(THTensor *tensor, THTensor *src, int dim, THLongTens
 
   index = THLongTensor_newContiguous(index);
   index_data = THLongTensor_data(index);
+  bool src_is_contiguous = src->is_contiguous();
+  if (!src_is_contiguous) {
+    src = THTensor_(newContiguous)(src);
+  }
 
-  if (dim == 0 && THTensor_(isContiguous)(src) && THTensor_(isContiguous)(tensor))
+  if (dim == 0 && THTensor_(isContiguous)(tensor))
```

Comment:
You can get better performance by moving all ` src_is_contiguous` related code **into** this `if` branch.

Also feel free to simplify the code by:
```cpp
src = THTensor_(newContiguous)(src);
// do something with src
c10::raw::intrusive_ptr::decref(src);
```
as `THTensor_(newContiguous)(src)` doesn't copy already contiguous tensors.

Reply:
Yes, code has been refined 
Thanks